### ISSUE TITLE
Unused config false-positive when using strict mode and maps

### ISSUE THREAD

**Issue by utybo** [sksamuel/hoplite#320]
I'm not entirely sure of what's the cause for this, but I'm getting an unused report in the following scenario. ~~May be related to the usage of maps in TOML files?~~ It's not, see my next comment.

```kotlin
import com.sksamuel.hoplite.ConfigLoaderBuilder
import com.sksamuel.hoplite.toml.TomlPropertySource

data class MapContainer(
    val myMap: Map<String, ExampleValue> = emptyMap()
)

data class ExampleValue(
    val number: Int = 1
)

val toml = """
[myMap.one]
number = 1
"""

val config = ConfigLoaderBuilder.default()
    .strict()
    .addPropertySource(TomlPropertySource(toml))
    .build()
    .loadConfigOrThrow<MapContainer>()

println(config)
```

This fails with

```
Exception in thread "main" com.sksamuel.hoplite.ConfigException: Error loading config because:

    Config value 'myMap.one' at (:2:1) was unused
	at com.sksamuel.hoplite.ConfigLoader$returnOrThrow$1.invoke(ConfigLoader.kt:191)
	at com.sksamuel.hoplite.ConfigLoader$returnOrThrow$1.invoke(ConfigLoader.kt:188)
	at com.sksamuel.hoplite.fp.ValidatedKt.getOrElse(Validated.kt:114)
	at com.sksamuel.hoplite.ConfigLoader.returnOrThrow(ConfigLoader.kt:188)
	at org.jetbrains.kotlin.idea.scratch.generated.ScratchFileRunnerGenerated$ScratchFileRunnerGenerated.<init>(tmp.kt:58)
	at org.jetbrains.kotlin.idea.scratch.generated.ScratchFileRunnerGenerated.main(tmp.kt:29)
```

Note that, if I remove the strict mode, everything is parsed normally without errors.

A possible workaround would be to disable the strict mode, but I'd like to avoid that if possible.

**Comment by utybo** [comment]
Seems that I can also repro this using YAML, so this does not seem to be related to TOML:

```kotlin
val yaml = """
myMap:
  one:
    number: 1
"""

val config = ConfigLoaderBuilder.default()
    .strict()
    .addPropertySource(YamlPropertySource(yaml))
    .build()
    .loadConfigOrThrow<MapContainer>()
```

This results in the same exception.

**Comment by utybo** [comment]
I've made a PR with a potential fix #321
